indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -180,6 +182,25 @@ class ObjectIdentifier @Throws(Asn1Exception::class) private constructor(
                     byteArrayOf((first().shortValue() * 40 + get(1).shortValue()).toUByte().toByte())
                 ) { i, acc, bytes -> if (i >= 2) acc + bytes else acc }
         }
+
+
+        private val customOidDescriptions = mutableMapOf<ObjectIdentifier, String>()
+
+        /**
+         * Adds or overrides a description of an Object Identifier. This is useful for communicating context to humans, especially for (but not limited to) debugging.
+         * Most well-known OIDs are already described in [OidMap].
+         * This method is neither thread-safe not coroutine-safe! Unguarded concurrent calls can cause loss of descriptions.
+         * OID descriptions need to live outside the actual OID objects, because this semantic enhancement will never be serialized and thus cannot be deserialized.
+         */
+        fun ObjectIdentifier.describe(description: String) = apply { customOidDescriptions[this] = description }
+
+        /**
+         * Returns a human-readable description of this OID. Virtually all commonly used OIDs will have this method pull a description from [KnownOIDs]/[OidMap].
```

Comment:
but only if you called the magic method first, right?

Reply:
true. needs s cross-ref here